### PR TITLE
Avoid bare 'new' in almost all other places.

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

**Comment by bangerth** [comment]
Part of #2375.

File: source/postprocess/particles.cc [hunk 1]
```diff
@@ -199,11 +199,10 @@ namespace aspect
 
           // Create the temporary file; get at the actual filename
           // by using a C-style string that mkstemp will then overwrite
-          char *tmp_filename_x = new char[tmp_filename.size()+1];
-          std::strcpy(tmp_filename_x, tmp_filename.c_str());
-          int tmp_file_desc = mkstemp(tmp_filename_x);
-          tmp_filename = tmp_filename_x;
-          delete []tmp_filename_x;
+          std::unique_ptr<char[]> tmp_filename_x (new char[tmp_filename.size()+1]);
```
Comment (tjhei) [inline source/postprocess/particles.cc]:
```suggestion
          std::vector<char> tmp_filename_x (tmp_filename.size()+1);
```
?

File: source/postprocess/visualization.cc [hunk 2]
```diff
@@ -677,11 +677,10 @@ namespace aspect
 
           // Create the temporary file; get at the actual filename
           // by using a C-style string that mkstemp will then overwrite
-          char *tmp_filename_x = new char[tmp_filename.size()+1];
-          std::strcpy(tmp_filename_x, tmp_filename.c_str());
-          int tmp_file_desc = mkstemp(tmp_filename_x);
-          tmp_filename = tmp_filename_x;
-          delete []tmp_filename_x;
+          std::unique_ptr<char[]> tmp_filename_x (new char[tmp_filename.size()+1]);
```
Comment (tjhei) [inline source/postprocess/visualization.cc]: same as above

**Comment by bangerth** [comment]
Good point -- check out the current state!

**Comment by tjhei** [comment]
thanks!

**Pull request merged**